cmd/messaging-tool/receive.go

```diff
@@ -42,23 +45,21 @@ func callback(m client.Message, topic string) (err error) {
 }
 
 func runReceive(cmd *cobra.Command, args []string) {
-	var c client.Connection
+	var c client.Client
 
 	// Check mandatory arguments:
 	if destinationName == "" {
 		glog.Errorf("The argument 'destination' is mandatory")
 		return
 	}
 
-	c = &stomp.Connection{
-		// Global options:
-		BrokerHost:   brokerHost,
-		BrokerPort:   brokerPort,
-		UserName:     userName,
-		UserPassword: userPassword,
-		UseTLS:       useTLS,
-		InsecureTLS:  insecureTLS,
-	}
+	// Set the clients variables before we can open it.
+	c.SetBrokerHost(brokerHost).
+		SetBrokerPort(brokerPort).
+		SetUserName(userName).
+		SetUserPassword(userPassword).
+		SetUseTLS(useTLS).
+		SetInsecureTLS(insecureTLS)
```

Comment:
Setter methods don't usually have the `Set` prefix in Go.

What happens if once the connection is created I call `connection.SetBrokerHost(123)` will it change the port immediately? Will it change it the next time I sent a message? Will it be completely ignored? I think that these methods don't belong in this type; they should be in a different type that is used to create the connection, and discarded after that.